Engine/DataFeeds/Enumerators/LiveEquityDataSynchronizingEnumerator.cs

```diff
@@ -120,7 +111,7 @@ public void Dispose()
         private bool DataPointEmitted(DateTime frontierUtc)
         {
             // check if any enumerator is ready to emit
-            if (_auxDataEnumeratorAdvanced && _auxDataEnumerator.Current != null && _tradeBarEnumeratorAdvanced && _tradeBarAggregator.Current != null)
+            if (_auxDataEnumerator.Current != null && _tradeBarAggregator.Current != null)
```

Comment:
When do we receive aux data in live mode? I'm thinking of a symbol change or a dividend event. IIRC, they come in almost 24 hours before they need to be emitted on the feed, BICBW, they might come down just-in-time.

I believe this was trying to support a case where we get aux data but we must wait for time to elapse before emitting it, and while waiting for time to elapse, we are still emitting from the `_tradeBarAggregator`. It looks like this behavior has been removed through removal of the enumerator advanced flags.

If possible, we should spin up some test algorithms with equities with known upcoming aux data events (dividends are the easiest) running at various data resolutions complete with some good logging.

Reply:
Yes, I believe the only aux data we currently have in live mode is split and dividend warnings and events, you wrote the scraper 😄 
Totally agree that the enumerator should emit the aux data only at the right time but the current version is not emitting if it comes in too early (i.e. the data has a timestamp greater than the current frontier).
The bug can be verified easily by running the existing unit test and counting the data points emitted (it will only emit 2 instead of 15):
https://github.com/QuantConnect/Lean/blob/master/Tests/Engine/DataFeeds/Enumerators/LiveEquityDataSynchronizingEnumeratorTests.cs#L31
With the fix, the updated test is now passing.

Reply:
I suppose my real concern is if we emit this data at the wrong time then it can cause issues with live synchronization as it's `EndTime` will be used as a frontier. This will cause the live sync to wait until the end time is reach and if there's other data sitting behind it and it's not sorted, then we'll miss out on that data. Consider the following:

00:00 6.25.2019 we _receive_ aux data w/ end time 00:00 6.26.2019
Since we're emitting immediately, it will go down to live sync, where it will wait until 00:00 6.26.2019.
For the entire day of 6.25.2019, we'll continue to receive pricing data for that symbol, but since the aux data is blocking that enumerator due to frontier time checks, we won't receive any data on 6.25.2019 for that symbol. Once the clock ticks over to 6.26.2019, the aux data will get pushed through _and then_ all the pricing data from 6.25.2019 with timestamps in the past would then finally get pushed through all as a single `DataFeedPacket`

Is my analysis above incorrect? Were we able to run a bunch of equity algorithms with known upcoming splits/dividends events and _prove_ that the above case is not happening?

Reply:
> 00:00 6.25.2019 we receive aux data w/ end time 00:00 6.26.2019

Unless I'm missing something, this aux data point should not be emitted until 00:00 6.26.2019, 
given the end time frontier check in L120 and L140, right?
While the "early" aux data point is waiting for the end of day 25, price data should still be emitted on L131?

I will follow your suggestion and confirm this with an equity algorithm with an upcoming dividend/split.